components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/management/PreUpdatePasswordActionConverter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.pre.update.password.action.management;
+
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.ActionDTO;
+import org.wso2.carbon.identity.action.management.service.ActionConverter;
+import org.wso2.carbon.identity.certificate.management.model.Certificate;
+import org.wso2.carbon.identity.user.pre.update.password.action.model.PasswordSharing;
+import org.wso2.carbon.identity.user.pre.update.password.action.model.PreUpdatePasswordAction;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.user.pre.update.password.action.constant.PreUpdatePasswordActionConstants.CERTIFICATE;
+import static org.wso2.carbon.identity.user.pre.update.password.action.constant.PreUpdatePasswordActionConstants.PASSWORD_SHARING_FORMAT;
+
+/**
+ * This class implements the methods required to build Action objects in Pre Update Password extension.
+ */
+public class PreUpdatePasswordActionConverter implements ActionConverter {
+
+    @Override
+    public Action.ActionTypes getSupportedActionType() {
+
+        return Action.ActionTypes.PRE_UPDATE_PASSWORD;
+    }
+
+    @Override
+    public ActionDTO buildActionDTO(Action action) {
+
+        PreUpdatePasswordAction preUpdatePasswordAction = (PreUpdatePasswordAction) action;
+        PasswordSharing passwordSharing = preUpdatePasswordAction.getPasswordSharing();
+
+        Map<String, Object> properties = new HashMap<>();
+        if (passwordSharing != null && passwordSharing.getFormat() != null) {
```

Comment:
Should we check for both null or empty or against a specified list.

Reply:
Fixed